src/Core/Services/ApiService.cs

```diff
@@ -593,7 +593,8 @@ public Task<PasswordlessLoginResponse> GetAuthResponseAsync(string id, string ac
 
         public Task<PasswordlessLoginResponse> PostCreateRequestAsync(PasswordlessCreateLoginRequest passwordlessCreateLoginRequest, AuthRequestType authRequestType)
         {
-            return SendAsync<object, PasswordlessLoginResponse>(HttpMethod.Post, authRequestType == AuthRequestType.AdminApproval ? "/auth-requests/admin-request" : "/auth-requests", passwordlessCreateLoginRequest, authRequestType == AuthRequestType.AdminApproval, true);
+            return SendAsync<object, PasswordlessLoginResponse>(HttpMethod.Post, authRequestType == AuthRequestType.AdminApproval ? "/auth-requests/admin-request" : "/auth-requests", passwordlessCreateLoginRequest, authRequestType == AuthRequestType.AdminApproval, true,
+                (message) => message.Headers.Add("Device-Identifier", passwordlessCreateLoginRequest.DeviceIdentifier));
```

Comment:
🤔 Is it `Device-Identifier` or `X-Device-Identifier` the proper header to add? Because I see [this](https://github.com/bitwarden/mobile/blob/main/src/Core/Services/ApiService.cs#L405C15-L405C15) on `GetKnownDevice` and has an `X-` as a prefix. if we should use the same, please extract this into a general constant that can be reused.

Reply:
It's `Device-Identifier`, server side the `GetKnownDevice` is using an inline method annotation `FromHeader` [link](https://github.com/bitwarden/server/blob/main/src/Api/Controllers/DevicesController.cs#L209).
Here i'm taking advantage of user context. [link](https://github.com/bitwarden/server/blob/main/src/Core/Context/CurrentContext.cs#L71).

Reply:
In both cases it is a header, I was wondering if we should always use the same name to be consistent, if it's referring to the same thing. Maybe you changed the name not to fire some logic, if that's the case, it's fine.

Reply:
I had to use `Device-Identifier` to use server CurrentContext functionality to exclude the current device from receiving the notification .  
The `SendMessageAsync` has a parameter to exclude current context. [link](https://github.com/bitwarden/server/blob/main/src/Core/Services/Implementations/NotificationsApiPushNotificationService.cs#L193)